document-store/src/main/java/org/hypertrace/core/documentstore/query/Query.java

```diff
@@ -105,13 +111,50 @@ public static QueryBuilder builder() {
       return Collections.emptyList();
     }
 
-    return sort.getSortingSpecs();
+    return Collections.unmodifiableList(sort.getSortingSpecs());
   }
 
   public Optional<Pagination> getPagination() {
     return Optional.ofNullable(pagination);
   }
 
+  @AllArgsConstructor
+  public static final class TransformedQueryBuilder {
```

Comment:
Good point. I was in dilemma as well. If we take this class outside this package, we may have to make the Query constructor public. Moreover, we may need to provide getters for all the fields.

So, I thought, it's better to keep this "friend" class here so that any query construction cannot happen outside of the Query class. Considering this as a special type of builder of the Query class, so that we do not have to expose the Query constructor outside (since we also need to perform the hibernate validations on top).

Reply:
Still not comfortable exposing this internals to the clients interacting with the Query class. 
Prefer extracting this out to a separate class within the query transformer package. Instead of making the query constructor public, I believe we can internally use the Query builder again. We anyways have exposed a getter for all the fields

Reply:
Makes sense. I have moved this class to a separate query transformer package (outside of the mongo package).